### PR TITLE
Support cloud user and nebula native user can be mixed

### PR DESCRIPTION
**Pull request by bright-starry-sky**
related the new feature for cloud user and nebula native user. issue #2911 .

Please hold this PR. we need both front-end and back-end debugging.

### CODE REVIEW

File: src/graph/service/GraphService.cpp [hunk 1]
```diff
@@ -164,12 +164,21 @@ bool GraphService::auth(const std::string& username, const std::string& password
...
+    // Cloud user and native user will be mixed.
+    // Since cloud user and native user has the same transport protocol,
+    // There is no way to identify which one is in the graph layer，
+    // let's check the native user's password first, then cloud user.
+    auto pwdAuth = std::make_unique<PasswordAuthenticator>(queryEngine_->metaClient());
+    if (pwdAuth->auth(username, encryption::MD5Utils::md5Encode(password))) {
+      return true;
+    }
+    auto cloudAuth = std::make_unique<CloudAuthenticator>(queryEngine_->metaClient());
+    return cloudAuth->auth(username, password);
```
Comment (kqzh) [inline src/graph/service/GraphService.cpp:181]: so in cloudAuth->auth function, there is no need to checkShadowAccount now?

**Comment by Shylock-Hg** [comment]
I think better to add one `field` in request to identify them

**Comment by bright-starry-sky** [comment]
And you need to delete the god user "root" first before create cloud root user.

**Comment by veezhang** [comment]
Would it be better to authorize explicitly rather than implicitly?

For example, use a code code, or add a field to indicate explicitly that the cloud allows password authentication.

```c++

bool CloudAuthenticator::auth(const std::string& user, const std::string& password) {
  // ...
  try {
    auto json = folly::parseJson(result.value());
    if (json["code"].asString().compare("0") != 0) {
      LOG(ERROR) << "Cloud authentication failed, user: " << user;
      return false;
    }

    // password_authentication eq true indicates that cloud allows use the password authentication
    if (json["password_authentication"].asString().compare("true") == 0) {
      auto authenticator = std::make_unique<PasswordAuthenticator>(metaClient_);
      return authenticator->auth(user, encryption::MD5Utils::md5Encode(password));
    }
  } catch (std::exception& e) {
    LOG(ERROR) << "Invalid json: " << e.what();
    return false;
  }
  return true;
}
```

**Review by panda-sheep — APPROVED** [review]
Good job

**Review by kqzh — APPROVED** [review]
LGTM

**Pull request merged**